### PR TITLE
Chapter 3: Subgroups

### PR DESCRIPTION
**Pull request by MAO3J1m0Op**
Depends on the merging of chapters 0, 1 and 2.

### CODE REVIEW

**Comment by MAO3J1m0Op** [comment]
We are now waiting on the merge of chapters 1 and 2. The PR is now open for reviewing, but I will leave it as a draft until chapter 2 is merged and we have cleaned up our chapter correspondingly (as we will need some more reviewing on those changes).

**Comment by Stavan-Jain** [comment]
image

There are a bunch of definitions that the blueprint alludes to but do not exist in the lean files

File: AlgebraInLean/Chapter03/Sheet01.lean [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+variable {G : Type*} [Group G]
+
+/-- This instance coerces `Subgroup G` to `Set G` by extracting the underlying set. -/
+instance : Coe (Subgroup G) (Set G) := ⟨λ H ↦ H.carrier⟩
+/-
+This instance permits the use of `H : Subgroup G`. An element `a : H`, will have two properties:
+`a.val`, which is of type `G`, and `a.property`, which is the hypothesis that `a.val ∈ H`. Under the
+hood, this uses the built in `Subtype` type.
+-/
+instance : CoeSort (Subgroup G) (Type _) := ⟨λ H ↦ H.carrier⟩
```
Comment (Stavan-Jain) [inline AlgebraInLean/Chapter03/Sheet01.lean:51]: What is `Type _`?

File: AlgebraInLean/Chapter03/Sheet02.lean [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+end Mpow
+
+section Gpow
+
+/--
+Now, we define the power function for groups. Since groups have inverses, there becomes a natural
+notion of negative exponentiation. Notice that `Int` has two constructors.
+-/
+def gpow {G : Type*} [Group G] (x : G) : ℤ → G
+/- `Int.ofNat` covers the positive end of the integers. -/
```
Comment (Stavan-Jain) [inline AlgebraInLean/Chapter03/Sheet02.lean]: Could you explain this in more detail?

File: AlgebraInLean/Chapter03/Sheet02.lean
Comment (Stavan-Jain) [inline AlgebraInLean/Chapter03/Sheet02.lean:1]: Nice!!

**Comment by MAO3J1m0Op** [comment]
We're gonna merge even with the `sorry`s in sheet 3. For this reason, we need #42; otherwise the blueprint and documentation on GitHub pages will not update.

**Review by Stavan-Jain — APPROVED** [review]
lgtm

**Review by MithicSpirit — APPROVED** [review]
probably good enough. too late for me to care

**Review by justin-t-morrill — APPROVED** [review]
cool

**Review by 4rim — APPROVED** [review]
letm (looks epic to me)

**Pull request merged**